From: Jianchao Wang <jianchao.w.wang@oracle.com>
Subject: [PATCH] blk-mq: return when BLK_MQ_S_STOPPED in blk_mq_run_work_fn
Date: Mon,  4 Jun 2018 17:03:55 +0800
Cc: linux-block@vger.kernel.org, linux-kernel@vger.kernel.org
To: axboe@kernel.dk, ming.lei@redhat.com

if hctx is stopped, don't run the queue in blk_mq_run_work_fn.

Fixes: 15fe8a9 (blk-mq: remove blk_mq_delay_queue())
Cc: Ming Lei <ming.lei@redhat.com>
Signed-off-by: Jianchao Wang <jianchao.w.wang@oracle.com>
---
 block/blk-mq.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/block/blk-mq.c b/block/blk-mq.c
index 9ce9cac..d0ee928 100644
--- a/block/blk-mq.c
+++ b/block/blk-mq.c
@@ -1578,7 +1578,7 @@ static void blk_mq_run_work_fn(struct work_struct *work)
 	 * If we are stopped, don't run the queue.
 	 */
 	if (test_bit(BLK_MQ_S_STOPPED, &hctx->state))
-		clear_bit(BLK_MQ_S_STOPPED, &hctx->state);
+		return;
 
 	__blk_mq_run_hw_queue(hctx);
 }
